File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-202808-59615/workdir/src/main/java/surveycrawler/NormalRun.java
package surveycrawler;

public class NormalRun extends CrawlerRun {

    // guarded run

    @Override
    public void startGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.beginGuardedRun();
    }

    @Override
    public void endGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.signalNotStartedGuardedRun();
    }

    @Override
    public void assertIsFinished(SurveyCrawler aCrawler) {
        // a normal run is always finished
    }

    // moving

    @Override
    public void slideOnSilt(SurveyCrawler aCrawler, Point aDirection, SiltSlide aSlide) {
        aCrawler.moveBy(aDirection.times(aSlide.cellsToSlide()));
    }

    // undoing

    @Override
    public void registerUndoAction(Runnable anUndoAction) {
        // movements of a normal run are not undone
    }

    @Override
    public void handleErrorOf(SurveyCrawler aCrawler) {
        // movements of a normal run are not undone
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/own-tests-project/src/surveycrawler/normal_run.py
from surveycrawler.crawler_run import CrawlerRun


class NormalRun(CrawlerRun):

    # guarded run

    def start_guarded_run(self, a_crawler):
        a_crawler.begin_guarded_run()

    def end_guarded_run(self, a_crawler):
        a_crawler.signal_invalid_command()

    def finish(self, a_crawler):
        pass

    # undoing

    def register_undo(self, an_undo_action):
        pass

    def handle_error(self, a_crawler):
        pass

    # moving

    def slide_crawler_towards(self, a_crawler, a_direction, a_silt):
        a_number_of_cells = a_silt.number_of_cells_to_slide()
        a_crawler.move_to(a_crawler.position().plus(a_direction.times(a_number_of_cells)))
